sdk/storage/Azure.Storage.Files.DataLake/src/DataLakePathClient.cs

```diff
@@ -712,6 +712,304 @@ internal virtual void SetNameFieldsIfNull()
+        /// </returns>
+        /// <remarks>
+        /// A <see cref="RequestFailedException"/> will be thrown if
+        /// a failure occurs.
+        /// </remarks>
+        private async Task<Response<PathInfo>> CreateIfNotExistsInternal(
+            PathResourceType resourceType,
+            PathHttpHeaders httpHeaders,
+            Metadata metadata,
+            string permissions,
+            string umask,
+            bool async,
+            CancellationToken cancellationToken)
+        {
+            DiagnosticScope scope = ClientDiagnostics.CreateScope($"{nameof(DataLakePathClient)}.{nameof(Create)}");
+            Response<PathInfo> response;
+            try
+            {
+                scope.Start();
+                DataLakeRequestConditions conditions = new DataLakeRequestConditions { IfNoneMatch = new ETag("*") };
+                response = await CreateInternal(
+                    resourceType,
+                    httpHeaders,
+                    metadata,
+                    permissions,
+                    umask,
+                    conditions,
+                    async,
+                    cancellationToken).ConfigureAwait(false);
+            }
+            catch (RequestFailedException storageRequestFailedException)
+            when (storageRequestFailedException.ErrorCode == "PathAlreadyExists")
+            {
+                response = default;
+            }
+            catch (Exception ex)
+            {
+                scope.Failed(ex);
+                throw;
+            }
+            finally
+            {
+                scope.Dispose();
+            }
+            return response;
+        }
+        #endregion Create If Not Exists
+
+        #region Exists
+        /// <summary>
+        /// The <see cref="Exists"/> operation can be called on a
+        /// <see cref="DataLakePathClient"/> to see if the associated
+        /// file or director exists in the file system.
+        /// </summary>
+        /// <param name="cancellationToken">
+        /// Optional <see cref="CancellationToken"/> to propagate
+        /// notifications that the operation should be cancelled.
+        /// </param>
+        /// <returns>
+        /// Returns true if the file or directory exists.
+        /// </returns>
+        /// <remarks>
+        /// A <see cref="RequestFailedException"/> will be thrown if
+        /// a failure occurs. If you want to create the file system if
+        /// it doesn't exist, use
+        /// <see cref="CreateIfNotExists"/>
+        /// instead.
+        /// </remarks>
+        public virtual Response<bool> Exists(
+            CancellationToken cancellationToken = default)
+        {
+            DiagnosticScope scope = ClientDiagnostics.CreateScope($"{nameof(DataLakePathClient)}.{nameof(Exists)}");
+
+            try
+            {
+                scope.Start();
+
+                return _blockBlobClient.Exists(
+                cancellationToken);
```

Comment:
Style nit - not sure the extra line buys much for a single parameter, but we should at least indent

Reply:
fixed.